src/NFTEDA.sol

```diff
@@ -62,57 +58,20 @@ contract ExponentialPriceDecayNFTAuction {
     /// @dev assumes the nft being sold is already controlled by the auction contract
     /// @param auction The defintion of the auction
     /// @return id the id of the auction
-    function startAuction(Auction calldata auction) external returns (uint256 id) {
-        id = auctionID(auction);
-
-        if (auctionStartTime[id] != 0) {
-            revert AuctionExists();
-        }
-
-        auctionStartTime[id] = block.timestamp;
-
-        emit StartAuction(
-            id,
-            auction.auctionAssetID,
-            auction.auctionAssetContract,
-            auction.perPeriodDecayPercentWad,
-            auction.secondsInPeriod,
-            auction.startPrice,
-            auction.paymentAsset
-            );
-    }
+    function startAuction(Auction calldata auction) external virtual returns (uint256 id);
```

Comment:
could have this method be shared between all subclasses? and just include a starter on the base Auction struct regardless of whether they receive a discount or not? seems would be nice to share starter functionality

Reply:
yeah, possibly should. I created separate because they store separate info and can simple one has an optimized check for if the auction already exists